tests/validation/tests/v3_api/test_pipeline.py

```diff
@@ -0,0 +1,107 @@
+from .common import get_global_admin_client_and_cluster, create_kubeconfig, \
+    random_test_name, create_project_and_ns, get_project_client_for_token, \
+    random_str, wait_for_condition, get_cluster_client_for_token, \
+    create_connection, WebsocketLogParse, CATTLE_TEST_URL, USER_TOKEN
+import pytest
+import urllib
+import time
+
+namespace = {"p_client": None, "ns": None, "cluster": None,
+             "project": None, "pipeline": None, "pipeline_run": None}
+PIPELINE_TIMEOUT =  600
+PIPELINE_REPO_URL = "https://github.com/rancher/pipeline-example-go.git"
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    client, cluster = get_global_admin_client_and_cluster()
+    create_kubeconfig(cluster)
+    p, ns = create_project_and_ns(
+        USER_TOKEN, cluster, random_test_name("testworkload"))
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+
+    def fin():
+        # project delete doesn't remove all ns in test, removing manually
+        client = get_cluster_client_for_token(namespace["cluster"], USER_TOKEN)
+        for ns_x in client.list_namespace(projectId=namespace["project"].id):
+            client.delete(ns_x)
+        client.delete(namespace["project"])
+    request.addfinalizer(fin)
+
+
+def check_last_run_state(status):
+    def _find_condition(resource):
+        if not hasattr(resource, "lastRunState"):
+            return False
+
+        if resource.lastRunState is None:
+            return False
+
+        if resource.lastRunState == status:
+            return True
+        if resource.lastRunState == "Failed":
+            return False
+        return False
+    return _find_condition
+
+
+def create_example_pipeline():
+    return namespace["p_client"].create_pipeline(
+        name="test-" + random_str(),
+        repositoryUrl=PIPELINE_REPO_URL,
+        triggerWebhookPr=False,
+        triggerWebhookPush=False,
+        triggerWebhookTag=False)
+
+
+def test_run_example_pipeline():
```

Comment:
Can we have this test in the top, following by @pytest.fixture and the helper methods?

Reply:
And can you rename this to test_pipeline ?